airflow/config_templates/airflow_local_settings.py

```diff
@@ -286,6 +287,17 @@
             },
         }
         DEFAULT_LOGGING_CONFIG["handlers"].update(OSS_REMOTE_HANDLERS)
+    elif REMOTE_BASE_LOG_FOLDER.startswith("webhdfs://"):
+        WEBHDFS_REMOTE_HANDLERS = {
+            "task": {
+                "class": "airflow.providers.apache.hdfs.log.webhdfs_task_handler.WebHDFSTaskHandler",
+                "formatter": "airflow",
+                "base_log_folder": os.path.expanduser(BASE_LOG_FOLDER),
+                "webhdfs_log_folder": REMOTE_BASE_LOG_FOLDER,
+                "filename_template": FILENAME_TEMPLATE,
+            }
+        }
+        DEFAULT_LOGGING_CONFIG["handlers"].update(WEBHDFS_REMOTE_HANDLERS)
```

Comment:
I'm not a huge fan of how we're creating these nested objects inside of an if/else statement. this feels like a bit of code rot. Is there a way to create a generator class for all of these?

Reply:
I agree with you, the current solution scales very poorly.

Before I try and implement a improvement, I'm not sure what direction we should take on this. I could create a class/function that does the same switch logic and centralizes the common parameters, `base_log_folder`, `REMOTE_BASE_LOG_FOLDER`, `FILENAME_TEMPLATE` and `formatter`. Is that what you mean?

Adding to that, I feel the issue here is more regarding the non-uniformity of the remote loggers, as I described in my other comment. If we are looking for a better uniform solution, I propose we discuss the remote logging architecture in an issue, as it would be a huge scope creep for this MR.